universal_resolver.go

```diff
@@ -0,0 +1,49 @@
+package ens
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/wealdtech/go-ens/v3/contracts/universalresolver"
+)
+
+type UniversalResolver struct {
+	backend      bind.ContractBackend
+	Contract     *universalresolver.Contract
+	ContractAddr common.Address
+}
+
+// NewUniversalResolver obtains the ENS UniversalResolver.
+func NewUniversalResolver(backend bind.ContractBackend) (*UniversalResolver, error) {
+	address, err := UniversalResolverContractAddress(backend, "mainnet")
```

Comment:
This will fail for non-mainnet instances.  The backend should be queried for the chain ID and a suitable address returned.